x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java

```diff
@@ -349,8 +349,10 @@ protected void onQueryFailure(int shardIndex, SearchShardTarget shardTarget, Exc
         }
 
         @Override
-        protected void onFetchFailure(int shardIndex, Exception exc) {
+        protected void onFetchFailure(int shardIndex, SearchShardTarget shardTarget, Exception exc) {
             checkCancellation();
+            searchResponse.get().addShardFailure(shardIndex,
+                new ShardSearchFailure(exc, shardTarget.getNodeId() != null ? shardTarget : null));
```

Comment:
maybe add a comment explaining why we have this != null?

Reply:
I added https://github.com/elastic/elasticsearch/pull/55688/commits/245199625184a56b58d3e0524ccb55958b6ba494